modules/netaddress.go

```diff
@@ -81,7 +94,14 @@ func (na NetAddress) IsValid() error {
 		if len(host) < 1 || len(host) > 253 {
 			return errors.New("invalid hostname length")
 		}
-		for _, label := range strings.Split(host, ".") {
+		if host[0] == '-' {
+			return errors.New("hostnames cannot start with a hyphen")
```

Comment:
as per the RFC, labels can't begin *or end* with a hyphen. I would use `if strings.HasPrefix(host, "-") || strings.HasSuffix(host, "-")`